Lib/test/test_pyrepl/test_pyrepl.py

```diff
@@ -748,3 +754,50 @@ def test_bracketed_paste_single_line(self):
         reader = self.prepare_reader(events)
         output = multiline_input(reader)
         self.assertEqual(output, input_code)
+
+
+class TestMain(TestCase):
```

Comment:
These are a bit heavy weight (starting a process, writing to it, polling, shutting the process down). While I don't think it makes sense for most of the tests to be written this way, I do think there is functionality that is hard to test without this sort of approach. For instance, to check the exposed globals, we could use `importlib.import_module` and then run `dir` on the imported module. However, that doesn't execute the `__main__` block, so it isn't comprehensive. There is similar tests for the old REPL in `Lib/test/test_repl`.